target: tizenrt-artik05x: Initial porting

glistening opened the pull request:
Initial porting to tizenrt-artik05x.

Tizen RT is a lightweight RTOS-based platform to support low-end IoT devices.
The project is under developing in github TizenRt repo.
Artik05x is an IoT SoC solution that will be released soon.

This PR contains the minimum porting that can print Hello
world from jerryscript and a document for building and running.

I used jerry_port.c and jerry_main.c in nuttx port and default port for
porting base code. Also I used Make.defs and Makefile in working
example in tizenrt for adding jerryscript configuration.

JerryScript-DCO-1.0-Signed-off-by: Sanggyu Lee sg5.lee@samsung.com

zherczeg commented on targets/tizenrt-artik05x/apps/jerryscript/Makefile:
```diff
@@ -0,0 +1,177 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
```
Is it possible to combine the first two licenses?

glistening replied:
I think both license are necessary although it is same Apache license, because the Makefile comes from TizenRT source and we usually keep the original license.

zherczeg commented on targets/tizenrt-artik05x/apps/jerryscript/jerry_main.c:
```diff
@@ -0,0 +1,326 @@
+static void
+print_help (char *name)
+{
+  jerry_port_console ("Usage: %s [OPTION]... [FILE]...\n"
```
The jerry_port_console is planned to be phased out from the project. I would recommend to use printf.

glistening replied:
Ok. I will get rids of it.

glistening replied:
I tried to remove `jerry_port_console`. However, `jerry-core/ecma/builtin-objects/ecma-builtin-global.c` uses `jerry_port_console`. Perhaps I can remove `jerry_port_console` after #1749 is merged.

zherczeg commented on targets/tizenrt-artik05x/jerry_port.c:
```diff
@@ -0,0 +1,121 @@
+int
+setjmp (jmp_buf buf)
+{
+  return __builtin_setjmp (buf);
```
The __builtin_setjmp implementation is incorrect on NuttX. I would recommend to use the solution from the NuttX port and add an assembly file for it.

glistening replied:
Thank you for informing me.  `setjmp/longjmp` implementation on NuttX is not yet checked correctly working in tizenrt-artik05x. I had a plan to make another patch. The next patch will contains like followings:
- Integrating `longjmp/setjmp` from NuttX port (after confirming correctly working on TizenRT)
- Adapting incoming breaking changes (e.g., removing `jerry_port_console`, `PORT_DIR`, ...)
...

LaszloLango commented:
Please check #1776 and #1777 
IMHO it would be better to land #1777 at first and follow the new design for very new port/target. @akosthekiss what do you think?

glistening commented:
@LaszloLango I rebased since #1777 landed.
@zherczeg I removed the uses of `jerry_port_console` in this PR. However, I have to provide the definition of `jerry_port_console` since it is required before #1749 is merged.

akosthekiss commented:
Sorry for the late review, but cannot find `jerry_port.c` in the port. (Or, do I miss something?)

yichoi commented:
Alas

yichoi commented:
Alas, It's initial porting. he will ramp up the Tizen RT features. :p

akosthekiss commented:
Fine. I was just puzzled as `jerry_port.c` was explicitly mentioned in the commit message and was also referenced in the makefiles.

glistening commented:
@akosthekiss Oh. It's my mistake! After moving ` jerry_port.c` from `PORT_DIR` to tizenrt app path. I didn't do `git add`. I will push the file within 4 hours.
